### ISSUE TITLE
Problem with preview in Gcam app

### ISSUE THREAD

**Issue by NooB9496** [deckerst/aves#1273]
**Describe the bug**
When I want check picture preview in Gcam, then aves gallery crash like this on video. 

**To Reproduce**
1. Go to 'Gcam '
2. Click on 'picture preview'
3. See error

**Expected behavior**

**System information and logs:**

Package: deckers.thibault.aves.libre
Installer: dev.imranr.obtainium
Aves version: 1.11.17-libre, build 13602
Flutter: stable 3.24.4
Android version: 14, API 34
Android build: UKQ1.230804.001 release-keys
Device: Xiaomi 2210132G
Support: dynamic colors=true, geocoder=true, HDR=true
Mobile services: not available
Connectivity: wifi, vpn
System locales: pl_PL
Storage volumes: /storage/emulated/0/
Storage grants: /storage/emulated/0/Pictures/, /storage/emulated/0/DCIM/, /storage/emulated/0/Android/media/
Error reporting: false

![aves-logs-20241103_035941.txt](https://github.com/user-attachments/files/17609075/aves-logs-20241103_035941.txt)

https://github.com/user-attachments/assets/44f70cc3-2f38-4c25-971c-7281c2754962

**Additional context**
Gcam version AGC9.2.14_V13.0_aweme. Aves Libre is set as a default Gallery app.

**Comment by NooB9496** [comment]
Before I use Fossify Gallery app and I don't have this problem.

**Comment by deckerst** [comment]
Maybe related: 
- #997 
- #781

**Comment by deckerst** [comment]
Do you use a custom ROM?

**Comment by deckerst** [comment]
I found some AGC9.2.14_V13.0_aweme but it seems a bit suspicious, like a GCam tweaked with some nagging to support the dev, and some outside package for LUTs. Does that sound like what you are using? Could you give me the URL of the site where you got your version?

**Comment by NooB9496** [comment]
https://www.celsoazevedo.com/files/android/google-camera/f/configs-bigkaka-07/

**Comment by NooB9496** [comment]
> Do you use a custom ROM?

Yes, Xiaomi eu rom.

**Comment by deckerst** [comment]
I've managed to reproduce the issue, with the following caveats:
-  as previously reported in #997, this issue happens only when the app is selected to open the GCam preview "Always". When it's "Just once", it works fine.
- I can always reproduce it with the `release` version of Aves, sometimes the `profile` version, and never the `debug` version.

**Comment by NooB9496** [comment]
> Maybe related:
> 
> * [Grey photo viewer after pressing "Always" in GCam #997](https://github.com/deckerst/aves/issues/997)
> * [GCAMs compatibility #781](https://github.com/deckerst/aves/issues/781)

Yeah, when I choose "just once" then aves works, but when I choose "always" then I have Grey photo viewer.

Soon I will try send logcat

**Comment by deckerst** [comment]
Thanks, but no need for you to send `logcat` logs. I can reproduce the issue, so I'm already checking the logs on my end.

**Comment by NooB9496** [comment]
> Thanks, but no need for you to send `logcat` logs. I can reproduce the issue, so I'm already checking the logs on my end. 

Too late 😁 ![logcat_11-04-2024_20-31-10.txt](https://github.com/user-attachments/files/17622988/logcat_11-04-2024_20-31-10.txt)

**Comment by NooB9496** [comment]
Maybe it's a problem with the android "chain-running" feature of the app?

**Comment by deckerst** [comment]
"chain-running"? Do you mean opening an app within another app, or something else?

**Comment by NooB9496** [comment]
> "chain-running"? Do you mean opening an app within another app, or something else?

Yes

**Comment by laundmo** [comment]
I'm experiencing this issue as well.

I noticed a curious thing: if i repeatedly tap my screen really fast right after clicking the gallery/preview button, sometimes Aves shows up.

Due to this,it feels like some sort of race condition.

**Comment by xusism** [comment]
I'm having this exact same issue on AGC 9.1 from BigKaka too, not happening with MGC 8.1 tho
